samples/corda/corda-simple-application/workflows-kotlin/src/main/kotlin/com/example/flow/AssetFlow.kt

```diff
@@ -624,3 +632,204 @@ class AcceptorOfTransfer(val otherPartySession: FlowSession) : FlowLogic<SignedT
+    }
+}
+
+/**
+ * The GetSimpleAssetStateAndContractId flow first checks if the JSON encoded object corresponds to the specified simple asset attribute values.
+ * It first unmarshalls the passed JSON encoded object and verifies if the attribte values match with the input values. Then, it creates
+ * the asset state object corresponding to the JSON object passed as input.
+ *
+ * @property marshalledAsset The JSON encoded fungible simple asset.
+ * @property type The fungible simple asset type.
+ * @property quantity The number of units of the fungible simple asset.
+ * @property lockerCert The owner (certificate in base64 of the exporting network) of the fungible asset before asset-transfer.
+ * @property holder The party that owns the fungible simple asset after asset-transfer.
+ */
+@InitiatingFlow
+@StartableByRPC
+class GetSimpleAssetStateAndContractId(
+    val marshalledAsset: String,
+    val type: String,
+    val quantity: Long,
+    val lockerCert: String,
+    val holder: Party
+): FlowLogic<Pair<String, AssetState>>() {
+    @Suspendable
+    override fun call(): Pair<String, AssetState> {
+
+        println("Inside GetTokenStateAndContractId().")
+
+        // must have used GsonBuilder().create().toJson() at the time of serialization of the JSON
+        val pledgedFungibleAsset = Gson().fromJson(marshalledAsset, AssetStateJSON::class.java)
+        println("Unmarshalled fungible simple asset is: $pledgedFungibleAsset")
+
+        if (pledgedFungibleAsset.tokenType != type) {
+            println("pledgedFungibleAsset.tokenType(${pledgedFungibleAsset.tokenType}) need to match with type(${type}).")
+            throw Exception("pledgedFungibleAsset.tokenType(${pledgedFungibleAsset.tokenType}) need to match with type(${type}).")
+        } else if (pledgedFungibleAsset.quantity != quantity) {
+            println("pledgedFungibleAsset.numUnits(${pledgedFungibleAsset.quantity}) need to match with quantity(${quantity}).")
+            throw Exception("pledgedFungibleAsset.numUnits(${pledgedFungibleAsset.quantity}) need to match with quantity(${quantity}).")
+        } else if (pledgedFungibleAsset.ownerCert != lockerCert) {
+            println("pledgedFungibleAsset.ownerCert(${pledgedFungibleAsset.ownerCert}) need to match with lockerCert(${lockerCert}).")
+            throw Exception("pledgedFungibleAsset.ownerCert(${pledgedFungibleAsset.ownerCert}) need to match with lockerCert(${lockerCert}).")
+        }
+
+        val simpleasset = AssetState(
+            pledgedFungibleAsset.quantity, // @property quantity
+            pledgedFungibleAsset.tokenType, // @property tokenType
+            holder // @property owner
+        )
+
+        return Pair(AssetContract.ID, simpleasset)
+    }
+}
+
+/**
+ * The GetOurCertificateBase64 flow is used to fetch the certificate of the party owning the node in base64 format.
+ */
+@InitiatingFlow
+@StartableByRPC
+class GetOurCertificateBase64() : FlowLogic<String>() {
```

Comment:
Check Slack for this, to remove such flows, and implement directly in client.

Reply:
Sandeep, `proxy.legalIdentitiesAndCerts.get(0).certificate` is not outputting the same certificate as `ourIdentityAndCert.certificate`, surprisingly! So, I have to retain this flow.

Reply:
Fixed the issue, now we can get Pem from Corda Cert, which is same in flow and client. Shared in slack.

Reply:
Update Sandeep.